src/lib/reasoners/arrays_rel.ml

```diff
@@ -258,52 +260,69 @@ let update_env are_eq are_dist dep env acc gi si p p_ded n n_ded =
     let conseq = LRmap.add n n_ded dep conseq in
     { env with split = sp; conseq = conseq }, acc
 
-  | Some _,  Some _ -> assert false
+  | Some _,  Some _ ->
+    (* This case cannot occur because we cannot have `gi = si` and
+       `gi <> si` at the same time. *)
+    assert false
+
+(* Produce a formula to propagate a read over a store.
```

Comment:
Same as the comment above about notation, we should settle one notation and not mix multiple ones (in comments use `read` and `write`, in the code we use `get` and `set`, and here we use `store` as well)